src/lockfree/list.c

```diff
@@ -179,14 +176,29 @@ int list_remove(llist_t *the_list, val_t val)
             return 0;
         }
         right_succ = right->next;
-        if (!is_marked_ref(right_succ)) {
+        if (!is_marked_ref((long)right_succ)) {
             if (CAS_PTR(&(right->next), right_succ,
-                        get_marked_ref(right_succ)) == right_succ) {
-                FAD_U32(&(the_list->size));
-                return 1;
+                        get_marked_ref((long)right_succ)) == right_succ) {
+                break;
             }
         }
     }
-    // we just logically delete it, someone else will invoke search and delete
-    // it
+    if (CAS_PTR(&(left->next), right, right_succ) == right) {
+        // physically delete
+        FAD_U32(&(the_list->size));
+        return 1;
+    }
+    // jusy logically delete
+    return 0;
+}
+
+void list_print(llist_t *the_list)
```

Comment:
You should split `list_print` (or similar changes) into separate commits. So that, we can track each functional changes.